public/app/shared/utilService.js

```diff
@@ -0,0 +1,172 @@
+/*!
+ * Copyright 2015 UC Berkeley (UCB) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+(function(angular) {
+
+  'use strict';
+
+  angular.module('collabosphere').service('utilService', function($location, $q) {
+
+    // Cache the API domain and Course ID that were passed in through
+    // the iFrame launch URL. These variables need to be used to construct
+    // the base URL for all REST API requests
+    var apiDomain = $location.search().api_domain;
```

Comment:
@nicolaasmatthijs am I correct in saying that you can't use the `$routeParams` here?